msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -283,36 +283,43 @@ To build am API-based message extension using Visual Studio Code, follow these s
 
  A new browser window with Teams web client opens. You can add your app to Teams.
 
-<!--# [Visual Studio](#tab/visual-studio)
+# [Visual Studio](#tab/visual-studio)
 
 Before you get started, ensure that you install the Visual Studio Enterprise 2022 Preview version 17.9.0 Preview 1.0, and install the **Microsoft Teams development tools** under **ASP.NET and web development** workload.
 
 To create an API-based message extension using Teams Toolkit for Visual Studio, follow these steps:
 
-1. Open Visual Studio.
+1. Open **Visual Studio**.
 1. Go to **File** > **New** > **Project...** or **New Project**.
 
 1. Search for **Teams** and select **Microsoft Teams App**.
 
+   :::image type="content" source="../assets/images/Copilot/api-based-me-vs-teams.png" alt-text="Screenshot shows the Microsoft Teams app option in Visual Studio.":::
+
+1. Enter the **Project name** and **Location**.
+1. Select **Create**.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-vs-new-app.png" alt-text="Screenshot shows the project name, Location, and Create option in Visual Studio.":::
+
 1. Select **Search Results from API**.
 
 1. Select any of the following options:
-   * Start with a new API
-   * Start with an OpenAPI Description
+   * If you want to start without an API, select **Start with a new API**.
+   * If you have an existing OpenAPI Description document, select **Start with an OpenAPI Description**.
 
 1. Select **Next**.
 
    :::image type="content" source="../assets/images/Copilot/api-based-me-vs-create-project.png" alt-text="Screenshot shows the Search results from API, New API,  OpenAPI Description Document, and Create options in Visual Studio to create a new Project.":::
 
 1. Enter OpenAPI specification URL or select **Browse..** to upload a file from your local machine.
 1. Under **Select Operations Copilot Can Interact with**, select dropdown and select APIs from the list.
-1. Select **Create**. The project is scaffolded and you can find API spec, manifest and response template files in the **appPackage** folder.
+1. Select **Create**. The project is scaffolded and you can find API specification, manifest, and response template files in the **appPackage** folder.
 
-   1. In the debug dropdown menu, select **Dev Tunnels** > **Create a Tunnel**.
+1. In the debug dropdown menu, select **Dev Tunnels** > **Create a Tunnel**.
```

Comment:
I think dev tunnel is only required if user chooses "start with a new API"

Reply:
updated with steps for start with a new API.